src/http/common.cr

```diff
@@ -82,7 +82,22 @@ module HTTP
     end
 
     io << "\r\n"
-    io << body if body
+
+    if body
+      if body.is_a?(IO)
+        if headers["Transfer-Encoding"] == "chunked"
```

Comment:
Well, the Response#initialize enforce that IO implies chunked, or the IO is consumed to body.
Maybe all the `if for_sending` should be moved from the constructor to the `HTTP#serialize_headers_and_body` and add the version as params.

Agree?